### PR TITLE
[StorageAction] onboard StorageAction with autorest

### PR DESCRIPTION
**Pull request by JoyerJin**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell. (⚠️**Target release** is a different concept from **API readiness**. Please click below links for details.)
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] No need for a release

- [ ] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

**Comment by blueww** [comment]
Please don't merge the PR before the swagger fix PR is merged: https://github.com/Azure/azure-rest-api-specs/pull/28718.

**Comment by JoyerJin** [comment]
> Please don't merge the PR before the swagger fix PR is merged: [Azure/azure-rest-api-specs#28718](https://github.com/Azure/azure-rest-api-specs/pull/28718).

Thank you. I have added the fix, used the commit on May 2, 2024. If it updates a new commit, please let me know.

**Comment by blueww** [comment]
The swagger PR is merged. 
This PR merge is not blocked by swagger now.

**Review by blueww — APPROVED** [review]

File: src/StorageAction/StorageAction.Autorest/docs/Get-AzStorageActionTasksReport.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+## EXAMPLES
+
+### Example 1: Get report
+```powershell
+Get-AzStorageActionTasksReport -ResourceGroupName joyer-test -StorageTaskName mytask1 | Format-List
+```
+
+```output
+FinishTime                   : 2024-02-28T10:16:58.9261483Z
+Id                           : /subscriptions/9e223dbe-3399-4e19-88eb-0975f02ac87f/resourceGroups/joyer-test/providers/Microsoft.StorageActions/storageTasks/mytask1/reports/instance1
```
Comment (YanaXu) [inline src/StorageAction/StorageAction.Autorest/docs/Get-AzStorageActionTasksReport.md:33]: hide the subscription id and resource group name or other names in examples.

File: src/StorageAction/StorageAction.Autorest/docs/Get-AzStorageActionTasksReport.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+ObjectFailedCount            : 0
+ObjectsOperatedOnCount       : 0
+ObjectsSucceededCount        : 0
+ObjectsTargetedCount         : 1
+RunResult                    : Succeeded
+RunStatusEnum                : Finished
+RunStatusError               : 0x0
+StartTime                    : 2024-02-28T10:07:53.0000000Z
+StorageAccountId             : /subscriptions/9e223dbe-3399-4e19-88eb-0975f02ac87f/resourceGroups/joyer-test/providers/Microsoft.Storage/storageAccounts/storagetasktest202402281
+SummaryReportPath            : https://storagetasktest202402281.blob.core.windows.net/rrrr/mytask1/testassign1/2024-02-28T10:08:00/SummaryReport.json
```
Comment (YanaXu) [inline src/StorageAction/StorageAction.Autorest/docs/Get-AzStorageActionTasksReport.md:44]: hide the storage name even if it does not exist.

File: src/StorageAction/StorageAction.Autorest/docs/New-AzStorageActionTask.md [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+CreationTimeInUtc            : 4/12/2024 9:56:05 AM
+Description                  : my storage task 3
+ElseOperation                : {{
+                                 "name": "DeleteBlob",
+                                 "onSuccess": "continue",
+                                 "onFailure": "break"
+                               }}
+Enabled                      : True
+Id                           : /subscriptions/11111111-2222-3333-4444-123456789101/resourceGroups/group001/providers/Microsoft.StorageActions/storageTasks/mytask3
+IdentityPrincipalId          : ea96114b-ac3c-4350-87f5-4db5a91c656c
```
Comment (YanaXu) [inline src/StorageAction/StorageAction.Autorest/docs/New-AzStorageActionTask.md:164]:
```suggestion
IdentityPrincipalId          : 11111111-2222-3333-4444-123456789101
```

File: src/StorageAction/StorageAction.Autorest/docs/New-AzStorageActionTask.md [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+IdentityTenantId             : 72f988bf-86f1-41af-91ab-2d7cd011db47
```
Comment (YanaXu) [inline src/StorageAction/StorageAction.Autorest/docs/New-AzStorageActionTask.md:165]:
```suggestion
IdentityTenantId             :11111111-2222-3333-4444-123456789101
```

**Review by YanaXu — APPROVED** [review]

**Pull request merged**